mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -124,9 +126,31 @@ def open_multifile_dataset(fileNames, calendar, simulationStartTime=None,
                                  startDate=startDate,
                                  endDate=endDate)
 
-    ds = xarray.open_mfdataset(fileNames,
-                               preprocess=preprocess_partial,
-                               decode_times=False, concat_dim='Time', **kwargs)
+    kwargs = {'decode_times': False,
+              'concat_dim': 'Time'}
+
+    softLimit = resource.getrlimit(resource.RLIMIT_NOFILE)[0]
+
+    # use autoclose if we will use more than half the soft limit on open files
+    autoclose = len(fileNames) > softLimit/2
+
+    try:
+        ds = xarray.open_mfdataset(fileNames,
+                                   preprocess=preprocess_partial,
+                                   autoclose=autoclose, **kwargs)
+    except TypeError as e:
+        if 'autoclose' in str(e):
+            if autoclose:
+                # This indicates that xarray version doesn't support autoclose
+                print 'Warning: open_multifile_dataset is trying to use autoclose=True but\n' \
+                      'it appears your xarray version doesn\'t support this argument. Will\n' \
+                      'try again without autoclose argument.'
+
```

Comment:
Note, if something goes wrong in xarray it could issue an error like https://github.com/pydata/xarray/pull/1198/files#diff-e7faa2e88465688f603e8f1f6d4db821R287.  

Do we want to cover those cases here?  The reason I say this is because it is possible, although hopefully unlikely because of our unit testing, there could be an xarray bug and it would be really good to catch this gracefully if possible.